app/code/Magento/CatalogSearch/Model/Config/Source/ListSort.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/**
+ * Catalog Search Product List Sortable allowed sortable attributes source
+ *
+ * @author     Magento Core Team <core@magentocommerce.com>
+ */
+namespace Magento\CatalogSearch\Model\Config\Source;
+
+class ListSort implements \Magento\Framework\Option\ArrayInterface
+{
+    /**
+     * Catalog config
+     *
+     * @var \Magento\Catalog\Model\Config
+     */
+    protected $catalogConfig;
+
+    /**
+     * Construct
+     *
+     * @param \Magento\Catalog\Model\Config $catalogConfig
+     */
+    public function __construct(\Magento\Catalog\Model\Config $catalogConfig)
+    {
+        $this->catalogConfig = $catalogConfig;
+    }
+
+    /**
+     * Retrieve option values array
+     *
+     * @return array
+     */
+    public function toOptionArray()
+    {
+        $options = [];
+        $options[] = ['label' => __('Relevance'), 'value' => 'relevance'];
+        foreach ($this->getCatalogConfig()->getAttributesUsedForSortBy() as $attribute) {
+            $options[] = ['label' => __($attribute['frontend_label']), 'value' => $attribute['attribute_code']];
+        }
+        return $options;
+    }
+
+    /**
+     * Retrieve Catalog Config Singleton
+     *
+     * @return \Magento\Catalog\Model\Config
+     */
+    protected function getCatalogConfig()
```

Comment:
Is there any reason for this to be a protected method. I would recommend:

1. Public if it is needed outside this class,
2. Private if only this class needs it,